SparkPost/2web2ui
# FE-830 - Implement Hibana app styling controlled by theme context

[FE-830](https://jira.int.messagesystems.com/browse/FE-830)

**NOTE:** This is branched from `FE-904` - this should _not_ be merged before that one.

### What Changed

- Adds additional vertical padding to the navigation component to help offset the presence of the Hibana controls
- Incorporates styles from the UX mock ups
- Incorporates `<HibanaBanner/>` component in to the `<Layout/>` component. Child components control the rendering of a stylesheet and some descriptive content
- Reformats build and start Node scripts with Prettier
- Adds new `copyMatchboxStyles` Node script and incorporates in to the build and start scripts
- Updates tests to meet ticket requirements
- Updates the Node versions Travis uses to `10.17` to match AWS and our `"engines"` config in the `package.json`

### How To Test

1. Sign up for a new account
1. Update the UI flags via the `account/control` endpoint as described in #1443 
1. The banner should initially look like this
![Screen Shot 2020-02-27 at 11 32 20 AM](https://user-images.githubusercontent.com/3613392/75464207-dd1cd280-5954-11ea-8a6c-9e7a865a33ce.png)
1. Verify that on page load, a network request for `styles-default.css` is made
1. Click 'Take a Look' and the banner should look like this:
![Screen Shot 2020-02-27 at 11 32 07 AM](https://user-images.githubusercontent.com/3613392/75464280-f9207400-5954-11ea-8618-b03029d0aa1b.png)
1. Verify that a request was made for `styles-hibana.css` (you should see some visual changes in the app as well)
1. Additionally, verify that when all subsections within the navigation are expanded, that navigation items are not blocked by the Hibana banner - this should be checked on multiple viewports

### To Do
- [X] Get theme toggling working in the dev env
- [x] Restyle `<HibanaControls/>` to look like the mocks from a color/typography POV
- [x] Verify that styled system props work
- [x] Set up production build webpack config stuff
- [x] Ensure both CSS files are _not_ requested unless the user swaps to the new design
- [x] Verify that the CSS files are overwritten on each build
- [x] Travis builds are failing due to invalid `package-lock.json`

---
